src/bvals/cc/bvals_cc_in_one.cpp

```diff
@@ -0,0 +1,553 @@
+                            const NeighborBlock &nb, MeshBlock *pmb) {
+  int cn = pmb->cnghost - 1;
+
+  IndexDomain interior = IndexDomain::interior;
+  const IndexShape &cellbounds = pmb->cellbounds;
+  si = (nb.ni.ox1 > 0) ? (cellbounds.ie(interior) - cn) : cellbounds.is(interior);
+  ei = (nb.ni.ox1 < 0) ? (cellbounds.is(interior) + cn) : cellbounds.ie(interior);
+  sj = (nb.ni.ox2 > 0) ? (cellbounds.je(interior) - cn) : cellbounds.js(interior);
+  ej = (nb.ni.ox2 < 0) ? (cellbounds.js(interior) + cn) : cellbounds.je(interior);
+  sk = (nb.ni.ox3 > 0) ? (cellbounds.ke(interior) - cn) : cellbounds.ks(interior);
+  ek = (nb.ni.ox3 < 0) ? (cellbounds.ks(interior) + cn) : cellbounds.ke(interior);
+
+  // send the data first and later prolongate on the target block
+  // need to add edges for faces, add corners for edges
+  if (nb.ni.ox1 == 0) {
+    if (nb.ni.fi1 == 1)
+      si += pmb->block_size.nx1 / 2 - pmb->cnghost;
+    else
+      ei -= pmb->block_size.nx1 / 2 - pmb->cnghost;
+  }
+  if (nb.ni.ox2 == 0 && pmb->block_size.nx2 > 1) {
+    if (nb.ni.ox1 != 0) {
+      if (nb.ni.fi1 == 1)
+        sj += pmb->block_size.nx2 / 2 - pmb->cnghost;
+      else
+        ej -= pmb->block_size.nx2 / 2 - pmb->cnghost;
+    } else {
+      if (nb.ni.fi2 == 1)
+        sj += pmb->block_size.nx2 / 2 - pmb->cnghost;
+      else
+        ej -= pmb->block_size.nx2 / 2 - pmb->cnghost;
+    }
+  }
+  if (nb.ni.ox3 == 0 && pmb->block_size.nx3 > 1) {
+    if (nb.ni.ox1 != 0 && nb.ni.ox2 != 0) {
+      if (nb.ni.fi1 == 1)
+        sk += pmb->block_size.nx3 / 2 - pmb->cnghost;
+      else
+        ek -= pmb->block_size.nx3 / 2 - pmb->cnghost;
+    } else {
+      if (nb.ni.fi2 == 1)
+        sk += pmb->block_size.nx3 / 2 - pmb->cnghost;
+      else
+        ek -= pmb->block_size.nx3 / 2 - pmb->cnghost;
+    }
+  }
+}
+
+// send boundary buffers with MeshBlockPack support
+// TODO(pgrete) should probaly be moved to the bvals or interface folders
+TaskStatus SendBoundaryBuffers(std::shared_ptr<MeshData<Real>> &md) {
+  Kokkos::Profiling::pushRegion("Task_SendBoundaryBuffers_MeshData");
+
+  auto var_pack = md->PackVariables(std::vector<MetadataFlag>({Metadata::FillGhost}));
+  const auto Nv = var_pack.GetDim(4);
+
+  Kokkos::Profiling::pushRegion("Reset boundaries");
+
+  int buffers_used = 0;
+  // reset buffer and count required buffers
+  for (int block = 0; block < md->NumBlocks(); block++) {
+    auto &rc = md->GetBlockData(block);
+    auto pmb = rc->GetBlockPointer();
+
+    for (auto &v : rc->GetCellVariableVector()) {
+      if (v->IsSet(parthenon::Metadata::FillGhost)) {
+        v->resetBoundary();
+      }
+    }
+
+    for (int n = 0; n < pmb->pbval->nneighbor; n++) {
+      parthenon::NeighborBlock &nb = pmb->pbval->neighbor[n];
+      // TODO(?) currently this only works for a single "Variable" per container.
+      // Need to update the buffer sizes so that it matches the packed Variables.
+      assert(rc->GetCellVariableVector().size() == 1);
```

Comment:
I just noticed that these asserts *always* fail, even for `advection` because there are three vectors. You actually want this to check for the number of variables with `Metadata::FillGhost` set.